001-time-format.md

```diff
@@ -0,0 +1,36 @@
+# Source Point Time Format
+
+## Summary
+
+Standardize the format of time information in source points to facilitate operations.
+
+## Motivation
+
+The time data of source points is stored according to its type:
+
+- Instant/snapshot points: hashmap with a single key "time";
+- Interval points: hashmap with keys "start" and "end".
+
+In order to operate on source points, it is required to either know in advance how the time information is stored or to [handle each different format](http://git.io/vf7GI).
+
+Operations such as sorting and selecting a subset of points could benefit from a standardized format.
+
+## Design
+
+The time information of source points can be reduced to a single timestamp plus some additional metadata:
+
+- Instant/snapshot points: just a timestamp, no additional metadata required;
+- Interval points: a single timestamp and a duration metadata.
+
+By giving a common name to the timestamp key of the time hashmap regardless of the source point type, it is possible to perform the operations described above without knowing the source point type in advance or handling all possible types.
```

Comment:
> This feels a little bit like cheating by using connascence of name. It's a little like saying, "we don't need polymorphism as long as we make sure that we name all our fields correctly!".

It's a decomposition of the time information. All representations of source point time have at least one timestamp that can be used for the basic operations required by the application. The remaining time data (or absence of it) associated with the point is mostly for presentation.